consumers/google.go

```diff
@@ -79,26 +79,31 @@ func (d *googleDNSConsumer) Sync(endpoints []*pkg.Endpoint) error {
 
 	change := new(dns.Change)
 
-	records := make([]*pkg.Endpoint, 0)
+	records := make(map[string][]*pkg.Endpoint)
 
 	for _, e := range endpoints {
 		record, exists := currentRecords[e.DNSName]
 
 		if !exists || exists && isResponsible(record.owner) {
-			records = append(records, e)
+			records[e.DNSName] = append(records[e.DNSName], e)
 		}
 	}
 
-	for _, svc := range records {
+	for dnsName, nested := range records {
+		ips := make([]string, 0, len(nested))
+		for _, svc := range nested {
+			ips = append(ips, svc.IP)
+		}
+
 		change.Additions = append(change.Additions,
 			&dns.ResourceRecordSet{
-				Name:    svc.DNSName,
-				Rrdatas: []string{svc.IP},
+				Name:    dnsName,
+				Rrdatas: ips,
 				Ttl:     300,
 				Type:    "A",
 			},
 			&dns.ResourceRecordSet{
-				Name:    svc.DNSName,
+				Name:    dnsName,
 				Rrdatas: []string{heritageLabel, labelPrefix + params.recordGroupID},
```

Comment:
Instead of a func I initialize the desired value in the producer's initializer. This way it's defined at a single place but it's not re-computed on every `sync` and `process`.